### DISCUSSION TITLE
Inquiry about External Configuration File Feature in Spotless Maven Plugin

### DISCUSSION THREAD

**Discussion by NihalJain** [diffplug/spotless#1799] [Q&A] [answered]
Dear Spotless Team,

I hope this message finds you well. I'm currently using the Spotless Maven Plugin in multiple Java projects to maintain a consistent code style. I have found it extremely helpful and efficient.

However, I came across a situation where I found it challenging to maintain the Spotless configurations consistently across different projects. I currently set the Spotless configurations in the <configuration> tag of the pom.xml file in each project.

While searching for solutions, it occurred to me that an ability to read Spotless configuration from an external XML file or equivalent could offer a more flexible way to share standard configurations across several projects. This would be particularly useful when projects don't share a common parent pom file or require specific or distinct formatting rules.

I wanted to ask if there's already a way to read Spotless configuration from an external file, which I might have missed in the documentation. If this feature is presently not part of the Spotless Maven Plugin, I believe it would be a valuable addition, enhancing both flexibility and standardization of the code styles. Also, please let me know if I should raise a feature request for same?

In case, if there's already an alternative way to achieve this or plans to incorporate this feature in the future, kindly let me know.

Thank you for your attention to this matter.

Best regards,
Nihal Jain

**Comment by nedtwigg** [comment]
I think you'll find the info you need here

- https://github.com/diffplug/spotless/issues/250

**Reply by NihalJain** [reply]
Actually, I am looking for something like this, similar to maven-checkstyle-plugin:
```
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-checkstyle-plugin</artifactId>
        <version>3.3.0</version>
        <configuration>
          <configLocation>checkstyle.xml</configLocation>
        </configuration>
      </plugin>
```
Sample source: https://maven.apache.org/plugins/maven-checkstyle-plugin/examples/custom-checker-config.html

Is something like this supported with spotless-maven-plugin?

**Reply by nedtwigg** [reply]
Not to my knowledge, but @lutovich might have some tips...

**Reply by lutovich** [reply]
Maven plugin currently supports only reading config files for specific formatters like `<eclipse>` or `<diktat>` but doesn't support reading the whole `<configuration>` section from a file.

This feels like a reasonable feature request. Spotless Maven Plugin could read configuration from a file (local or URL) and treat all other explicit configuration parameters as overrides.

**Reply by NihalJain** [reply, marked as answer]
Thank you @lutovich and @nedtwigg for your inputs. Raised feature request at https://github.com/diffplug/spotless/issues/1805 for further tracking.
